trinity/plugins/builtin/ethstats/ethstats_service.py

```diff
@@ -134,13 +134,19 @@ def get_node_block(self) -> EthstatsData:
         }
 
     async def get_node_stats(self) -> EthstatsData:
-        response: PeerCountResponse = await self.context.event_bus.request(
-            PeerCountRequest()
-        )
+
+        try:
+            peer_count = (await asyncio.wait_for(
```

Comment:
Since this is inside of a `BaseService` class , I think we need to do `self.wait(awaitable, timeout=0.5)` so that it properly respect any triggered cancellations.